File: web/src/api/learning/guide.ts
import { useMutation, useQuery } from "@tanstack/react-query";

import { request } from "@/utils/request";

const BASE = "/api/learning";

export interface GenerateGuideRequest {
  objective_id: string;
  markdown: string;
}

export interface GuideResult {
  summary: string;
  mastery_goals: string[];
  practice_points?: GuidePracticePoint[];
  reading_steps: string[];
  pitfalls: string[];
  self_check_questions: string[];
  evidence: string[];
  content_hash?: string;
  cached?: boolean;
}

export interface GuidePracticePoint {
  title: string;
  goal: string;
  evidence?: string[];
}

const api = {
  get: (objectiveId: string, contentHash: string) =>
    request.get<GuideResult | null>(`${BASE}/guide/${objectiveId}`, {
      params: { content_hash: contentHash },
    }),

  generate: (data: GenerateGuideRequest) =>
    request.post<GuideResult>(`${BASE}/guide/generate`, data),
};

export const guideKeys = {
  all: ["learning-guide"] as const,
  detail: (objectiveId: string, contentHash: string) =>
    [...guideKeys.all, objectiveId, contentHash] as const,
};

export function useGuideCache(objectiveId: string, contentHash: string) {
  return useQuery({
    queryKey: guideKeys.detail(objectiveId, contentHash),
    queryFn: () => api.get(objectiveId, contentHash),
    enabled: !!objectiveId && !!contentHash,
  });
}

export function useGenerateGuide() {
  return useMutation({
    mutationFn: (data: GenerateGuideRequest) => api.generate(data),
  });
}

export const guideApi = api;

File: server/infrastructure/llm/prompts/guide.go
package prompts

func GuidePrompt(input Input) string {
	input = normalizeInput(input)
	switch input.PresetKey {
	case DefaultPresetKey:
		return guideInstruction
	default:
		return guideInstruction
	}
}

const guideInstruction = `你是导学老师。你的任务是阅读用户提供的 Markdown 学习资料,结合当前小目标,告诉学习者本节真正要掌握什么。

只输出 JSON,不要任何额外文字,格式:
{"summary":"本节导学摘要","mastery_goals":["掌握目标"],"practice_points":[{"title":"本轮复述小点","goal":"这一次只需要讲清楚什么","evidence":["资料依据短摘"]}],"reading_steps":["阅读步骤"],"pitfalls":["易错点"],"self_check_questions":["自检问题"],"evidence":["资料依据原文短摘"]}

要求:
- 禁止输出 <think>、markdown 代码块、解释文字或草稿,顶层第一个字符必须是 {,最后一个字符必须是 };
- 字符串里如需写代码片段,必须正确转义双引号,例如用 \"hello\" 表示字符串字面量;
- 必须基于资料正文,不要只根据标题泛泛而谈;
- mastery_goals 要具体到本节内容,不要写"理解核心概念"这类空话;
- practice_points 要把整篇资料拆成适合一次复述的小点,每个小点只考一个认知点,通常 3-7 条;
- practice_points.goal 要明确本轮复述的边界,避免要求学习者一次讲完整篇资料;
- evidence 必须摘自资料原文或忠实压缩原文,用于证明你读过资料;
- 每个数组通常 3-5 条,单条简洁;
- 如果资料不足,明确说明资料不足并给出需要补充的内容。`
